core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -564,6 +564,11 @@ object AdminUtils extends Logging {
     topics.map(topic => fetchTopicMetadataFromZk(topic, zkUtils, cachedBrokerInfo))
   }
 
+  def fetchTopicMetadataFromZkWithProtocol(topics: Set[String], zkUtils: ZkUtils, protocol: SecurityProtocol = SecurityProtocol.PLAINTEXT): Set[MetadataResponse.TopicMetadata] = {
```

Comment:
Yes, that's why I suggested that you should remove the default argument (you can't have both overloads and default arguments for a given method name). So:

```scala

def fetchTopicMetadataFromZk(topics: Set[String], zkUtils: ZkUtils): Set[MetadataResponse.TopicMetadata] =
  fetchTopicMetadataFromZk(topics, zkUtils, SecurityProtocol.PLAINTEXT)

def fetchTopicMetadataFromZk(topics: Set[String], zkUtils: ZkUtils, protocol: SecurityProtocol): Set[MetadataResponse.TopicMetadata] = {
    val cachedBrokerInfo = new mutable.HashMap[Int, Broker]()
    topics.map(topic => fetchTopicMetadataFromZk(topic, zkUtils, cachedBrokerInfo, protocol))
  }

 private def fetchTopicMetadataFromZk(topic: String,
                                       zkUtils: ZkUtils,
                                       cachedBrokerInfo: mutable.HashMap[Int, Broker],
                                       protocol: SecurityProtocol): MetadataResponse.TopicMetadata = {
```

We could also add an overload for the single topic case, but since it's just a convenience method anyway, it doesn't seem worth it.